### PR TITLE
chore: Enable G601 rule for gosec

### PR DESCRIPTION
**Pull request by zak-pawel**
resolves https://github.com/influxdata/telegraf/issues/12955

### CODE REVIEW

**Review by powersj — APPROVED** [review]

File: plugins/inputs/vsphere/endpoint.go [hunk 1]
```diff
@@ -1204,7 +1204,8 @@ func (e *Endpoint) collectChunk(
 			continue
 		}
 		buckets := make(map[string]metricEntry)
-		for _, v := range em.Value {
+		for i := range em.Value {
+			v := em.Value[i]
```
Comment (srebhan) [inline plugins/inputs/vsphere/endpoint.go:1208]: I think here the better alternative is to change `populateTags` to not take a `performance.MetricSeries` pointer but a value
```diff
- func (e *Endpoint) populateTags(objectRef *objectRef, resourceType string, resource *resourceKind, t map[string]string, v *performance.MetricSeries) {
+ func (e *Endpoint) populateTags(objectRef *objectRef, resourceType string, resource *resourceKind, t map[string]string, v performance.MetricSeries) {
```

The value of `v` is not changed in that function so the for loop can stay as-is.

**Review by srebhan — COMMENTED** [review]
I do have two comments where I think we should really fix the code instead of "silencing" the warning. Some other spots are "false positives" in the sense that the data in the pointer is never actually modified. The mongodb code is a really ugly `reflect` hell, but copies the struct (and uses a pointer to get an `interface`). The collectd part is an artifact from the collectd interface.

Anyhow, I think we should fix the two spots commented and proceed with the other changes. What do you think @zak-pawel?

File: plugins/outputs/azure_monitor/azure_monitor.go [hunk 2]
```diff
@@ -278,7 +278,8 @@ func (a *AzureMonitor) Write(metrics []telegraf.Metric) error {
 	}
 
 	var body []byte //nolint:prealloc // There is no point in guessing the final capacity of this slice
-	for _, m := range azmetrics {
+	for i := range azmetrics {
+		m := azmetrics[i]
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/outputs/azure_monitor/azure_monitor.go]: Changed.

**Review by srebhan — APPROVED** [review]
Looks great to me. Thanks @zak-pawel for fixing those issues!

**Pull request merged**